src/sql/workbench/parts/connection/common/connectionProviderExtension.ts

```diff
@@ -13,12 +14,35 @@ import { deepClone } from 'vs/base/common/objects';
 import * as azdata from 'azdata';
 import * as resources from 'vs/base/common/resources';
 
+
 export interface ConnectionProviderProperties {
 	providerId: string;
 	displayName: string;
 	connectionOptions: azdata.ConnectionOption[];
 }
 
+class ResolveChecker {
+	private resolvedUsers = new Map<IExtensionDescription, boolean>();
+
+	public removeFromResolvedUsers(oldUser: IExtensionPointUser<any>) {
+		this.resolvedUsers.delete(oldUser.description);
+	}
+
+	public clearResolvedUsers() {
+		this.resolvedUsers.clear();
+	}
+
+	public addToResolvedUsers(newUser: IExtensionPointUser<any>) {
+		this.resolvedUsers.set(newUser.description, true);
+	}
+
+	public checkResolvedUsers(thisUser: IExtensionPointUser<any>): boolean {
+		return this.resolvedUsers.has(thisUser.description);
+	}
```

Comment:
This works as expected? description is an object so if the calling code isn't re-using the same IExtensionDescription objects then we won't correctly be checking this.

It would be safer to use one of the other values the description has that are primitives such as description.identifier.value

Reply:
It works so far, as passing in the IExtensionPointUser for comparison resulted in duplicated results